atoma-state/src/handlers.rs

```diff
@@ -1767,3 +1908,1027 @@ mod utils {
+    ///
+    /// # Example
+    ///
+    /// ```rust,ignore
+    /// let swap_used = 2 * 1024 * 1024 * 1024;  // 2GB swap used
+    /// let swap_total = 8 * 1024 * 1024 * 1024;  // 8GB total swap
+    /// let weight = 0.05;
+    ///
+    /// let score = compute_swap_ram_score(swap_used, swap_total, weight);
+    /// // Returns 0.0375 because:
+    /// // 1. Swap usage is 25% (2GB/8GB)
+    /// // 2. Available swap percentage is 75% (1.0 - 0.25)
+    /// // 3. Final score is 0.75 * 0.05 = 0.0375
+    /// ```
+    ///
+    /// # Notes
+    ///
+    /// - The function assumes valid input where `ram_swap_used` <= `ram_swap_total`
+    /// - The score is part of a larger node performance evaluation system
+    /// - Higher scores indicate better swap memory availability
+    /// - Swap memory usage is generally less critical than RAM usage, hence typically lower weights
+    #[allow(clippy::cast_precision_loss)]
+    fn compute_swap_ram_score(
+        ram_swap_used: i64,
+        ram_swap_total: i64,
+        swap_ram_score_weight: f64,
+    ) -> f64 {
+        let ram_swap_used_percentage = (ram_swap_used as f64) / (ram_swap_total as f64);
+        swap_ram_score_weight * (1.0 - ram_swap_used_percentage)
+    }
+
+    /// Calculates a normalized network performance score based on network traffic.
+    ///
+    /// This function computes a score based on the total network traffic (receive + transmit),
+    /// where lower traffic indicates better network availability. The score uses an inverse
+    /// relationship with traffic volume, scaled by the provided weight factor.
+    ///
+    /// # Arguments
+    ///
+    /// * `network_rx` - The network receive traffic in bytes
+    /// * `network_tx` - The network transmit traffic in bytes
+    /// * `network_score_weight` - A weight factor to scale the final score (typically between 0.0 and 1.0)
+    ///
+    /// # Returns
+    ///
+    /// Returns a weighted f64 score where:
+    /// - Higher scores indicate lower network utilization (better availability)
+    /// - Lower scores indicate higher network utilization
+    /// - The score is scaled by `network_score_weight`
+    ///
+    /// # Example
+    ///
+    /// ```rust,ignore
+    /// let rx = 1000;  // 1KB received
+    /// let tx = 2000;  // 2KB transmitted
+    /// let weight = 0.2;
+    ///
+    /// let score = compute_network_score(rx, tx, weight);
+    /// // Returns 0.066 because:
+    /// // 1. Base score = 1 / (1 + 1000 + 2000) = 0.333
+    /// // 2. Final score = 0.333 * 0.2 = 0.066
+    /// ```
+    ///
+    /// # Notes
+    ///
+    /// - Uses an inverse relationship to ensure higher traffic results in lower scores
+    /// - The denominator adds 1 to prevent division by zero
+    /// - The score is part of a larger node performance evaluation system
+    #[allow(clippy::cast_precision_loss)]
+    fn compute_network_score(network_rx: i64, network_tx: i64, network_score_weight: f64) -> f64 {
```

Comment:
I think we also need to consider the bandwidth of the node as well, higher traffic on wider bandwidths may still have greater capacity for accepting work. e.g.

```rust
fn compute_network_score(
    network_rx: i64,
    network_tx: i64,
    network_capacity: i64,  // New parameter
    network_score_weight: f64
) -> f64 {
    let total_usage = network_rx as f64 + network_tx as f64;
    let utilization = total_usage / network_capacity as f64;
    
    // Sigmoid function to smooth the score
    network_score_weight * (1.0 / (1.0 + (-5.0 * (1.0 - utilization)).exp()))
}
```

Reply:
For sure, but we are not collecting `network_capacity` as of now. We will need to do it in the future

Reply:
See https://github.com/atoma-network/atoma-node/issues/410

Reply:
And #232